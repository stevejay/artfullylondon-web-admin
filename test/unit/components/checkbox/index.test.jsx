import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import Checkbox from '_admin/components/checkbox'

it('should render correctly when checked', () => {
  const wrapper = shallow(
    <Checkbox checked error={null} touched={false} onChange={_.noop}>
      <div id='child' />
    </Checkbox>
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when not checked', () => {
  const wrapper = shallow(
    <Checkbox checked={false} error={null} touched={false} onChange={_.noop}>
      <div id='child' />
    </Checkbox>
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when has an error', () => {
  const wrapper = shallow(
    <Checkbox checked error='The Error' touched onChange={_.noop}>
      <div id='child' />
    </Checkbox>
  )

  expect(wrapper).toMatchSnapshot()
})
