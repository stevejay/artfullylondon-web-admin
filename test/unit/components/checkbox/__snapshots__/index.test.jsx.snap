// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when checked 1`] = `
<div
  className="src-components-checkbox-index__container"
>
  <input
    checked={true}
    className="src-components-checkbox-index__checkbox"
    id="checkbox_1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="src-components-checkbox-index__label"
    htmlFor="checkbox_1"
  >
    <div
      id="child"
    />
  </label>
  <FaCheck
    className="src-components-checkbox-index__checkmark"
  />
</div>
`;

exports[`should render correctly when has an error 1`] = `
<div
  className="src-components-checkbox-index__container-error src-components-checkbox-index__container"
>
  <input
    checked={true}
    className="src-components-checkbox-index__checkbox"
    id="checkbox_3"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="src-components-checkbox-index__label"
    htmlFor="checkbox_3"
  >
    <div
      id="child"
    />
  </label>
  <FaCheck
    className="src-components-checkbox-index__checkmark"
  />
</div>
`;

exports[`should render correctly when not checked 1`] = `
<div
  className="src-components-checkbox-index__container"
>
  <input
    checked={false}
    className="src-components-checkbox-index__checkbox"
    id="checkbox_2"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="src-components-checkbox-index__label"
    htmlFor="checkbox_2"
  >
    <div
      id="child"
    />
  </label>
</div>
`;
